onnxruntime/python/providers/tvm/ort.py

```diff
@@ -73,49 +73,64 @@ def onnx_compile(model_string,
-                        model_nhwc = tvm.relay.transform.EliminateCommonSubexpr()(model_nhwc)
-                        irmod = tvm.relay.transform.FoldConstant()(model_nhwc)
-                    lib = relay.build(irmod, target=target, target_host=target_host)
-        else:
-            with tvm.transform.PassContext(opt_level=opt_level):
-                lib = relay.build(irmod, target=target, target_host=target_host, params=params)
-    elif tuning_type == AUTO_TVM_TYPE:
-        with relay.build_config(opt_level=opt_level):
-            if tuning_logfile:
-                # print("Use tuning file from ", AUTO_TVM_TYPE, ": ", tuning_logfile)
+                        seq = tvm.transform.Sequential(
+                            [
+                                relay.transform.InferType(),
+                                relay.transform.ConvertLayout(desired_layouts),
+                                relay.transform.EliminateCommonSubexpr(),
+                                relay.transform.FoldConstant(),
+                            ]
+                        )
+                        irmod = seq(irmod)
+                    lib = get_tvm_executor(irmod, executor, target, target_host, params)
+        elif tuning_type == AUTO_TVM_TYPE:
+            with relay.build_config(opt_level=opt_level):
+                log.info("Use tuning file from ", AUTO_TVM_TYPE, ": ", tuning_logfile)
                 with autotvm.apply_history_best(tuning_logfile):
-                    # XXX: do not pass parameters to relay.build otherwise they will be inline into the module
-                    lib = relay.build(irmod, target_host=target_host, target=target)
-            else:
-                lib = relay.build(irmod, target_host=target_host, target=target)
+                    lib = get_tvm_executor(irmod, executor, target, target_host, params)
+        else:
+            log.error("Tuning log type {} is unsupported. ".format(tuning_type) +
+                      "Only {} and {} types are supported".format(ANSOR_TYPE, AUTO_TVM_TYPE))
+            return None
     else:
-        # TODO(vvchernov): replace prints by logger, but investigate ORT logging system for python before
-        # print is not commented out while it declares error
-        print("ERROR: Tuning log type {} is unsupported. ".format(tuning_type),
-              "Only {} and {} types are supported".format(ANSOR_TYPE, AUTO_TVM_TYPE))
+        with tvm.transform.PassContext(opt_level=opt_level):
+            lib = get_tvm_executor(irmod, executor, target, target_host, params)
+
+    if lib is None:
         return None
 
     ctx = tvm.device(target, 0)
-    m = graph_executor.GraphModule(lib["default"](ctx))
+    if executor == "vm":
+        m = tvm.runtime.vm.VirtualMachine(lib, ctx)
+    elif executor == "graph":
+        m = graph_executor.GraphModule(lib["default"](ctx))
+    else:
+        print("ERROR: Executor type {} is unsupported. ".format(executor),
```

Comment:
Why not raising an exception?

Reply:
I've answered on it earlier, see your first question